### PR TITLE
29 pdf statistics

### PR DESCRIPTION
**Pull request by krzysztofruszczynski**

### CODE REVIEW

File: src/FileFormat/ImageStatistics.php [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    public function identifyPageSize($pageSize, $returnPageSizeInMmIfNotMatched = true)
+    {
+        $pageSizeArray = explode('x', $pageSize);
+
+        return sprintf('%dpx x %dpx',
+                $pageSizeArray[0],
+                $pageSizeArray[1]
+                );
+    }
+}
```
Comment (SimonHeimberg) [inline src/FileFormat/ImageStatistics.php]: Why don't we put this and PdfStatistics into a single file?
Then https://github.com/EmchBerger/projectassistant/blob/19700cbff8a68a1224eedce6ccbfacd81b1015f5/src/AppBundle/CubePa/DocRepro/Processing/FileStatistics.php#L9 would be simpler or unneeded.

This file PdfImageStatistics/FileStatistics/PrintFileStatistics/... could identify the document type itself in `setFilename()` (what Processing/FileStatistics.php does now) and optionally get access to this property.

Comment (krzysztofruszczynski) [reply src/FileFormat/ImageStatistics.php]: The idea behind this separation was, that it is easier to read and change things, which are only for images or pdfs.

File: src/FileFormat/ImageStatistics.php [hunk 2]
```diff
@@ -4,7 +4,7 @@
 /**
  * Class handling statistics for image files (based on statistics for PDF files).
  */
-class ImageStatistics extends PdfStatistics
+class ImageStatistics extends PdfStatistics implements FileStatisticsInterface
```
[the remark this answers is not in the document]
Comment (krzysztofruszczynski) [reply src/FileFormat/ImageStatistics.php]: Right, it is for transparency reasons. We can in future extend from different class, but we must implement this interface.

**Comment by MartinRosserEB** [comment]
Some comments:

https://github.com/EmchBerger/projectassistant/blob/19700cbff8a68a1224eedce6ccbfacd81b1015f5/src/AppBundle/CubePa/DocRepro/Processing/FileStatistics.php#L67
--> shouldn't we throw an error (File not found)? This allows easily spotting when and why something goes wrong during high-level testing. Same for (File with extension XYZ cannot be handled):
https://github.com/EmchBerger/projectassistant/blob/19700cbff8a68a1224eedce6ccbfacd81b1015f5/src/AppBundle/CubePa/DocRepro/Processing/FileStatistics.php#L75

Also here:
https://github.com/EmchBerger/cube-common-bundle/blob/master/src/FileFormat/PdfStatistics.php#L138
--> This will be difficult to follow up if we cannot see the error message (lots of components involved here).

**Pull request merged**